client/state/sharing/keyring/actions.js

```diff
@@ -12,9 +17,11 @@ import {
 /**
  * Triggers a network request for a user's connected services.
  *
+ * @param {Function} successCallback Function that is called after a successful
+ *                                   request. Default: _.identity.
  * @return {Function} Action thunk
  */
-export function requestKeyringConnections() {
+export function requestKeyringConnections( successCallback = identity ) {
```

Comment:
@gwwar and @obenland it would do me well to read more into this and I'd be happy to in time.

it sounds like we have the following situation:
 - there is a type of API call can occur repeatedly
 - the first time it happens it needs to be different
 - a second call should not occur until after the first one has returned successfully

if this is the case then middleware _is_ definitely a good option (see #8606) but I don't think it would be necessary to get out there. most important would be containing the API call into its own module, then inside the module scope we can define our own local state to track the sequencing.

is this helpful?